caffe2/core/blob.h

```diff
@@ -24,6 +25,41 @@ inline bool BlobIsTensorType(const Blob& blob, DeviceType device_type) {
   return tensor && *tensor && tensor->GetDeviceType() == device_type;
 }
 
+inline void BlobSetTensor(Blob* blob, Tensor tensor) {
+  blob->Reset<Tensor>(new Tensor(std::move(tensor)));
```

Comment:
Disgusting! :P

Reply:
(what's disgusting is, moving a Tensor into a heap allocated Tensor to pass into Reset. I'm not sure you can fix this easily though.)

Reply:
We probably don't need this API right now since we are going with `Output(i, dims, options)`.

Reply:
This won't be used, I will remove it..